src/payment/strategies/amazon-pay/amazon-pay-payment-strategy.ts

```diff
@@ -210,4 +237,39 @@ export default class AmazonPayPaymentStrategy implements PaymentStrategy {
             })
         );
     }
+
+    private _processPaymentWith3ds(sellerId: string, referenceId: string, methodId: string, useStoreCredit: boolean, options: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        return new Promise((resolve, reject) => {
+            if (!this._window.OffAmazonPayments) {
+                return reject(new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized));
+            }
+
+            return this._window.OffAmazonPayments.initConfirmationFlow(
+                sellerId,
+                referenceId,
+                (confirmationFlow: AmazonPayConfirmationFlow) => {
+                    return this._store.dispatch(
+                        this._orderActionCreator.submitOrder({useStoreCredit}, options)
```

Comment:
For 3DS flow we need to first submit the order because in `remote-checkout` endpoint we have a new call to amazon that needs the `orderId`. 
For amazon, we cannot do the validation in the backend, this is because the flow with and without 3DS are both different, what does this means? If 3DS is not active (the current flow), in submitOrder is sent a payment variable with the `methodId`, then in BCapp there is a condition that checks if the variable was sent, that means that the payment should be done in PHP, then it will be redirected to the order confirmation. On the other hand, if 3DS is active, we have the next steps: 1) creation of an empty order, 2) initialize the payment on Amazon, 3) redirect and 4) process the payment.

In fact amazon behavior is completely different to the others payment providers for example amazon never reach `bigpay`, also you can review the discussion about the backend approach in [PR INT-1616](https://github.com/bigcommerce/bigcommerce/pull/30507)

Reply:
Thanks @rlopz for the explanation. Yes, I'm aware of the discussion in your other PR. I raised this because, ideally, we want to avoid introducing another snowflake. But as you mentioned, this requires a flow that's different to all the other payment methods we have currently. We can't avoid this unless we put in significant refactoring work. So I guess what you have here is an acceptable compromise.